### PR TITLE
7317 use data artifacts

### PR DESCRIPTION
**Pull request by gdicristofaro**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java [hunk 1]
```diff
@@ -391,6 +429,23 @@ public BlackboardArtifact getGenInfoArtifact() throws TskCoreException {
...
+	/**
+	 * Returns the OS Account associated with this content if this is a file.
+	 * Otherwise, returns null.
+	 *
+	 * @return The OS Account ID associated with this content if this is a file.
+	 *         Otherwise, returns null.
+	 *
+	 * @throws TskCoreException
+	 */
+	private Long getOsAccountId() throws TskCoreException {
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java:441]: I thought we weren't going to add this to Content?

Comment (gdicristofaro) [reply bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java:441]: I'll take this out and resolve the merge conflict.

File: bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java [hunk 2]
```diff
@@ -318,27 +319,60 @@ public int hashCode() {
...
+
+		long thisObjId = getId();
+		Long dsObjId = getDataSource() != null ? getDataSource().getId() : null;
+
+		switch (artifactType.getCategory()) {
+			case DATA_ARTIFACT:
+				return this.newDataArtifact(artifactType, Collections.emptyList());
+			case ANALYSIS_RESULT: {
+				try {
+					AnalysisResultAdded addedResult = db.getBlackboard().newAnalysisResult(
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java:348]: Is there a reason we don't simply call this.newAnalysisResult() here?

File: bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java [hunk 3]
```diff
@@ -509,13 +510,22 @@ private Account getOrCreateAccount(Account.Type accountType, String accountUniqu
...
+
 		BlackboardArtifact accountArtifact = getAccountFileInstanceArtifact(accountType, accountUniqueID, sourceFile);
 		if (accountArtifact == null) {
-			accountArtifact = db.newBlackboardArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_ACCOUNT, sourceFile.getId());
-			Collection<BlackboardAttribute> attributes = new ArrayList<>();
-			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_ACCOUNT_TYPE, moduleName, accountType.getTypeName()));
-			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_ID, moduleName, accountUniqueID));
-			accountArtifact.addAttributes(attributes);
+			long fileObjId = sourceFile.getId();
+			Long dsObjId = sourceFile.getDataSource() == null ? null : sourceFile.getDataSource().getId();
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java:520]: It doesn't look like fileObjId or dsObjId are used

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 4]
```diff
@@ -5082,10 +5083,36 @@ String addSourceToArtifactAttribute(BlackboardAttribute attr, String source) thr
...
+	@Deprecated
 	public BlackboardArtifact newBlackboardArtifact(int artifactTypeID, long obj_id) throws TskCoreException {
 		BlackboardArtifact.Type type = getArtifactType(artifactTypeID);
-		return newBlackboardArtifact(artifactTypeID, obj_id, type.getTypeName(), type.getDisplayName());
+		if (type == null) {
+			throw new TskCoreException("Unknown artifact type for id: " + artifactTypeID);
+		}
+		
+		Category category = type.getCategory();
+		if (category == null) {
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:5096]: I don't believe category can be null

**Review by APriestman — APPROVED** [review]

File: bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java [hunk 5]
```diff
@@ -318,42 +319,60 @@ public int hashCode() {
...
 
+	@Deprecated
 	@Override
 	public BlackboardArtifact newArtifact(int artifactTypeID) throws TskCoreException {
 		// don't let them make more than 1 GEN_INFO
 		if (artifactTypeID == ARTIFACT_TYPE.TSK_GEN_INFO.getTypeID()) {
 			return getGenInfoArtifact(true);
 		}
-		return db.newBlackboardArtifact(artifactTypeID, objId);
+		BlackboardArtifact.Type artifactType = db.getArtifactType(artifactTypeID);
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java:329]: We should make SleuthkitCase.getArtifactType() do a lookup in the in the standard types map before doing a query. I will write a story about this.

Comment (APriestman) [reply bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java:329]: It looks like SleuthkitCase.getArtifactType() has caching and gets initialized with all the standard types

**Pull request merged**